### PR TITLE
Adding the private CA key to the K8S templates

### LINKED ISSUE #825: Adding the private CA key to the K8S templates
**Is this a request for help?**: No

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): Feature

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)** Kubernetes

This is needed for TLS cert signing in the API Server

### PR DESCRIPTION
**Pull request by anhowe**
**What this PR does / why we need it**: This is needed for TLS cert signing in the API Server and will help out with E2E conformance tests.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #825

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/827)

### CODE REVIEW

File: parts/kubernetesmastercustomscript.sh [hunk 1]
```diff
@@ -65,6 +66,19 @@ else
     echo "APISERVER_PRIVATE_KEY is empty, assuming worker node"
 fi
 
+# If CA_PRIVATE_KEY is empty, then we are not on the master
+if [[ ! -z "${CA_PRIVATE_KEY}" ]]; then
+    echo "CA_KEY is non-empty, assuming master node"
+
+    CA_PRIVATE_KEY_PATH="/etc/kubernetes/certs/ca.key"
+    touch "${CA_PRIVATE_KEY_PATH}"
+    chmod 0644 "${CA_PRIVATE_KEY_PATH}"
```
Comment (colemickens) [inline parts/kubernetesmastercustomscript.sh]: This should be `0600`.

**Comment by brendandburns** [comment]
Other than Cole's comment, LGTM.

Thanks!

**Comment by weinong** [comment]
@colemickens @anhowe  should we start adding CA private key to api model? as far as I can tell, this change will not work if we regenerate the template in RP.

**Comment by colemickens** [comment]
@weinong +1 I agree. For other bring-your-own-PKI, it wouldn't matter; but since we want to actually place the private key, I agree, it has to be part of the apimodel. (it already is, but isn't publicly exposed/persisted)

**Comment by anhowe** [comment]
Thanks @weinong, @colemickens this is now added.

**Pull request merged**